indico/modules/events/editing/operations.py

```diff
@@ -78,17 +78,25 @@ def _make_editable_files(editable, files):
 
 
 @no_autoflush
-def create_new_editable(contrib, type_, submitter, files):
+def create_new_editable(contrib, type_, submitter, files, initial_state=InitialRevisionState.ready_for_review):
     editable = Editable(contribution=contrib, type=type_)
     revision = EditingRevision(submitter=submitter,
-                               initial_state=InitialRevisionState.ready_for_review,
+                               initial_state=initial_state,
                                files=_make_editable_files(editable, files))
     editable.revisions.append(revision)
     db.session.flush()
     logger.info('Editable [%s] created by %s for %s', type_.name, submitter, contrib)
     return editable
 
 
+def delete_editable(editable):
+    db.session.delete(editable)
+    for revision in editable.revisions:
+        for ef in revision.files:
+            ef.file.unclaim()
```

Comment:
This is a bit dangerous: Even though it's never the case here (for now), a file could be referenced by other revisions etc. 

So I'm not sure if it's worth adding this just to avoid orphaned files. If we really *do* want to avoid orphaned files, I think we need to use something more sophisticated, like a reference counter (+1 on claim, -1 on unclaim) or check if there are any incoming relationship references left...

Reply:
The thing is, if I do not un-claim the files, the next submissions after an error will fail (e.g. when re-trying).

Reply:
OK good point. Maybe we can restrict this function to only work on an editable with a single revision? We don't expect to ever use it for other editables anyway. And since we do not allow previously-claimed files when creating a new editable, this would ensure you never have a file claimed elsewhere.

Could even go one step further and do the unclaiming in the RH itself and not in this operation..

Reply:
Not sure how I feel about having this in the RH itself since I expect to control this kind of operations in the file model. I'm leaving it as-is for now, perhaps later I could add a single revision check.